pkg/machine/e2e/cp_test.go

```diff
@@ -0,0 +1,172 @@
+		// ~/<ginkgo_tmp>
+		//   * foo.txt
+		//   - foo-dir
+		//     * bar.txt
+		//
+		//   * guest-foo.txt
+		//   - guest-foo-dir
+		//     * bar.txt
+
+		// GUEST FILE SYSTEM
+		// ~/
+		//   * foo.txt
+		//   - foo-dir
+		//     * bar.txt
+
+		var (
+			file     = "foo.txt"
+			filePath = filepath.Join(GinkgoT().TempDir(), file)
+
+			directory     = "foo-dir"
+			directoryPath = filepath.Join(GinkgoT().TempDir(), directory)
+
+			fileInDirectory     = "bar.txt"
+			fileInDirectoryPath = filepath.Join(directoryPath, fileInDirectory)
+
+			guestToHostFile = "guest-foo.txt"
+			guestToHostDir  = "guest-foo-dir"
+		)
+
+		_, err := os.Create(filePath)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = os.MkdirAll(directoryPath, 0755)
+		Expect(err).ToNot(HaveOccurred())
+
+		_, err = os.Create(fileInDirectoryPath)
+		Expect(err).ToNot(HaveOccurred())
+
+		name := randomString()
+		initMachine := initMachine{}
+		cp := cpMachine{}
+		sshMachine := sshMachine{}
+
+		By("host file to guest")
+		session, err := mb.setName(name).setCmd(initMachine.withImage(mb.imagePath).withNow()).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(0))
+
+		// copy the file into the guest
+		session, err = mb.setCmd(cp.withQuiet().withSrc(filePath).withDest(name + ":~/" + file)).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(0))
+
+		// verify the guest has the file
+		session, err = mb.setName(name).setCmd(sshMachine.withSSHCommand([]string{"ls"})).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(0))
+		Expect(session.outputToString()).To(Equal(file))
+
+		// try to copy the file to a location in the guest where permission will get denied
+		session, err = mb.setCmd(cp.withQuiet().withSrc(filePath).withDest(name + ":/etc/tmp.txt")).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(125))
+		Expect(session.errorToString()).To(ContainSubstring("scp: dest open \"/etc/tmp.txt\": Permission denied"))
+
+		By("host directory to guest")
+		// copy contents into the guest
+		session, err = mb.setCmd(cp.withQuiet().withSrc(directoryPath).withDest(name + ":~/" + directory)).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(0))
+
+		// verify the content is in the guest
+		session, err = mb.setName(name).setCmd(sshMachine.withSSHCommand([]string{"ls", directory})).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(0))
+		Expect(session.outputToString()).To(Equal(fileInDirectory))
+
+		By("guest file to host")
+		// copy contents to the host
+		guestToHostFilePath := filepath.Join(GinkgoT().TempDir(), guestToHostFile)
+		session, err = mb.setCmd(cp.withQuiet().withSrc(name + ":~/" + file).withDest(guestToHostFilePath)).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(0))
+		// check the contents are on the host
+		Expect(guestToHostFilePath).To(BeARegularFile())
+
+		// this test can only be run on Unix systems. Windows does not
```

Comment:
You can create a read-only folder with the following function that uses golang.org/x/sys/windows functions:

```go
func mkdirReadOnlyWindows(dirPath string) {
	pathPtr, err := windows.UTF16PtrFromString(dirPath)
	Expect(err).ToNot(HaveOccurred())

	// https://learn.microsoft.com/en-us/windows/win32/secauthz/security-descriptor-string-format
	dacl := "D:(A;;GRGX;;;WD)"
	sd, err := windows.SecurityDescriptorFromString(dacl)
	Expect(err).ToNot(HaveOccurred())

	sa := &windows.SecurityAttributes{
		Length:             uint32(unsafe.Sizeof(windows.SecurityAttributes{})),
		SecurityDescriptor: sd,
		InheritHandle:      1,
	}

	err = windows.CreateDirectory(pathPtr, sa)
	Expect(err).ToNot(HaveOccurred())
}
```

Reply:
I had brought this up in a meeting, and I had decided that as long as the other platforms pass this check, then it should be fine. I wanted to avoid having to break out a separate *_windows.go file to prove that something doesn't work as expected.

Is that ok with you? If not, I can absolutely do that work.

Reply:
I agree. This test doesn't seem significant enough to justify the complexity of using `CreateDirectory`.